app/controllers/podcast_metrics_controller.rb

```diff
@@ -3,11 +3,98 @@ class PodcastMetricsController < ApplicationController
 
   def show
     authorize @podcast, :show?
+
+    @episodes =
+      @podcast.episodes
+        .published
+        .order(published_at: :desc)
+        .paginate(params[:page], params[:per])
+    @date_start = 30.days.ago
+    @date_end = Time.zone.now
+
+    if clickhouse_connected?
+      @recent_downloads_by_episode =
+        Rollups::HourlyDownload
+          .where(podcast_id: @podcast.id, episode_id: @episodes.pluck(:guid), hour: (@date_start..@date_end))
+          .select(:episode_id, "DATE_TRUNC('DAY', hour) AS hour", "SUM(count) AS count")
+          .group(:episode_id, "DATE_TRUNC('DAY', hour) AS hour")
+          .order(Arel.sql("DATE_TRUNC('DAY', hour) DESC"))
+      @alltime_downloads_by_episode =
+        Rollups::HourlyDownload
+          .where(podcast_id: @podcast.id, episode_id: @episodes.pluck(:guid))
+          .select(:episode_id, "SUM(count) AS count")
+          .group(:episode_id)
+
+      @recent_downloads_total =
+        Rollups::HourlyDownload
+          .where(podcast_id: @podcast.id, hour: (@date_start..@date_end))
+          .select("SUM(count) AS count")
+          .group(:podcast_id)
+          .order(:podcast_id)
+      @alltime_downloads_total =
+        Rollups::HourlyDownload
+          .where(podcast_id: @podcast.id)
+          .select("SUM(count) AS count")
+          .group(:podcast_id)
+          .order(:podcast_id)
+
+      # @top_subdivs =
+      #   Rollups::DailyGeo
+      #     .where(podcast_id: @podcast.id)
+      #     .select(:country_code, :subdiv_code, "DATE_TRUNC('WEEK', day) AS day", "SUM(count) AS count")
+      #     .group(:country_code, :subdiv_code, "DATE_TRUNC('WEEK', day) AS day")
+      #     .order(Arel.sql("SUM(count) AS count DESC"))
+      #     .limit(10)
+      # @top_countries =
+      #   Rollups::DailyGeo
+      #     .where(podcast_id: @podcast.id)
+      #     .select(:country_code, "SUM(count) AS count")
+      #     .group(:country_code)
+      #     .order(Arel.sql("SUM(count) AS count DESC"))
+      #     .limit(10)
+      @agent_apps_query =
```

Comment:
3 queries for agent name, type, and os. below i also add pagination for apps and os so that they are not limited to top 10.